tools/topology/topology2/cavs/cavs-rt5682.conf

```diff
@@ -0,0 +1,240 @@
+<searchdir:cavs>
+<searchdir:include>
+<searchdir:include/common>
+<searchdir:include/components>
+<searchdir:include/dais>
+<searchdir:include/pipelines/cavs>
+<searchdir:cavs/platform/intel>
+
+<vendor-token.conf>
+<manifest.conf>
+<pdm_config.conf>
+<tokens.conf>
+<virtual.conf>
+<passthrough-playback.conf>
+<passthrough-capture.conf>
+<passthrough-be.conf>
+<data.conf>
+<pcm.conf>
+<pcm_caps.conf>
+<fe_dai.conf>
+<ssp.conf>
+<dmic.conf>
+<hda.conf>
+<hw_config.conf>
+<manifest.conf>
+<route.conf>
+<cavs/common_definitions.conf>
+<copier.conf>
+<pipeline.conf>
+<dai.conf>
+<host.conf>
+<dmic-default.conf>
+<hdmi-default.conf>
+
+Define {
+	MCLK 				24576000
+	NUM_DMICS			0
+	SPK_ID				7
+	# override DMIC default definitions
+	DMIC_IO_CLK			38400000
+	NUM_DMICS			0
+	DMIC0_ID			1
+	DMIC1_ID			2
+	DMIC0_NAME			'dmic01'
+	PDM1_MIC_A_ENABLE		1
+	PDM1_MIC_B_ENABLE		1
+	DMIC0_HOST_PIPELINE_ID		13
+	DMIC0_DAI_PIPELINE_ID		14
+	DMIC0_HOST_PIPELINE_SINK	'copier.host.13.1'
+	DMIC0_DAI_PIPELINE_SRC		'copier.DMIC.14.1'
+	DMIC0_PCM_CAPS			'Passthrough Capture 13'
+	DMIC0_PIPELINE_STREAM_NAME	'copier.DMIC.14.1'
+	# override HDMI default definitions
+	NUM_HDMIS			0
+	HDMI1_ID			3
+	HDMI2_ID			4
+	HDMI3_ID			5
+	HDMI4_ID			6
+	PLATFORM 			"none"
+}
+
+# override defaults with platform-specific config
+IncludeByKey.PLATFORM {
+	"mtl"	"platform/intel/mtl.conf"
+}
+
+# include DMIC config if needed.
+IncludeByKey.NUM_DMICS {
+	"[1-4]"	"platform/intel/dmic-generic.conf"
+}
+
+# include HDMI config if needed.
+IncludeByKey.NUM_HDMIS {
+"[3-4]" "platform/intel/hdmi-generic.conf"
+}
+
+#
+# List of all DAIs
+#
+#SSP Index: 0, Direction: duplex
+Object.Dai {
+	SSP."0" {
+		id 		0
+		dai_index	0
+		direction	"duplex"
+		name		SSP0-Codec
+		default_hw_conf_id	0
+		sample_bits		32
+		io_clk		$MCLK
+
+		Object.Base.hw_config."SSP0" {
+			id	0
+			mclk_freq	$MCLK
+			bclk_freq	3072000
+			tdm_slot_width	32
+		}
+	}
+	SSP."1" {
+		id 		$SPK_ID
+		dai_index	1
+		direction	"playback"
+		name		SSP1-Codec
+		default_hw_conf_id	0
+		sample_bits		32
+		io_clk		$MCLK
+
+		Object.Base.hw_config."SSP1" {
+			id	0
+			mclk_freq	$MCLK
+			bclk_freq	3072000
+			tdm_slot_width	32
+		}
+	}
+}
+
+#
+# Pipeline definitions
+#
+
+# Pipeline ID:1 PCM ID: 0
+Object.Pipeline {
+	# playback pipelines
+	passthrough-playback.1 {
+		index	1
+		Object.Widget.pipeline.1.stream_name	"passthrough-playback-ssp0"
+		Object.Widget.copier.1.stream_name	"SSP0 Playback"
+	}
+
+	passthrough-be.1 {
+		index	2
+		direction	playback
+
+		Object.Widget.pipeline.1.stream_name	"passthrough-playback-ssp0"
+
+		Object.Widget.copier.1 {
+			dai_index 0
+			dai_type "SSP"
+			copier_type "SSP"
+			stream_name "SSP0-Codec"
+			node_type $I2S_LINK_OUTPUT_CLASS
+		}
+	}
+
+	passthrough-playback.2 {
+		index	3
+		Object.Widget.pipeline.1.stream_name	"passthrough-playback-ssp1"
```

Comment:
For all MTL-based products, we want the ability to support multiple streams for playback and capture. So we cannot have any references to passthrough any longer. We need a clear split between host and dai pipelines, so that additional host pipelines can be added at a later point.

Reply:
Ok, then let me digest above, thanks.